python/pyiceberg/table/__init__.py

```diff
@@ -90,3 +103,90 @@ def snapshot_by_name(self, name: str) -> Optional[Snapshot]:
     def history(self) -> List[SnapshotLogEntry]:
         """Get the snapshot history of this table."""
         return self.metadata.snapshot_log
+
+
+class TableScan:
+    _always_true: ClassVar[BooleanExpression] = AlwaysTrue()
```

Comment:
This was needed to avoid calling functions in arg defaults.

Reply:
Not sure if this is necessary. `AlwaysTrue` is a singleton it will always return the same instance.